gdx/src/com/badlogic/gdx/graphics/g2d/GlyphLayout.java

```diff
@@ -183,19 +182,19 @@ else if (targetWidth <= fontData.spaceXadvance * 3) //
 					} else
 						lastGlyph = run.glyphs.peek();
 
-					if (!wrap || n == 0) { // No wrap or truncate, or no glyphs.
+					if (!wrapOrTruncate || xAdvancesSize == 0) { // No wrap or truncate, or no glyphs.
 						if (markupEnabled) { // If disabled any subsequent run is sure to be on the next line.
-							for (int i = 0; i < n; i++)
+							for (int i = 0; i < xAdvancesSize; i++)
 								x += xAdvances[i];
 						}
 						break runEnded;
 					}
 
 					// Wrap or truncate.
 					x += xAdvances[0]; // X offset relative to the drawing position.
-					for (int i = 1; i < n; i++) {
+					for (int i = 1; i < xAdvancesSize; i++) {
 						Glyph glyph = run.glyphs.get(i - 1);
-						float glyphWidth = (glyph.width + glyph.xoffset) * fontData.scaleX - fontData.padRight;
+						float glyphWidth = getGlyphWidth(glyph, fontData);
```

Comment:
The same code of `getGlyphWidth` in several places in the code.
Having it as a private mthod reduces errors and makes it easier to read imho.
Besides, I can be sure that this line does not change any variables.